webapp/src/toolbox.tsx

```diff
@@ -452,8 +452,16 @@ export class Toolbox extends data.Component<ToolboxProps, ToolboxState> {
         this.props.parent.onToolboxFocusCapture();
     }
 
-    handleBlurCapture = () => {
-        this.props.parent.onToolboxBlurCapture();
+    handleBlurCapture = (e: React.FocusEvent<HTMLDivElement>) => {
+        // Make sure we close the flyout if the search input is focussed and there 
+        // is no flyout for search blocks.
+        const searchInputEl = (this.refs.searchbox as React.Component<ToolboxSearch>).refs.searchInput;
```

Comment:
Well, this is certainly something.

Reply:
I think this whole block could do with a revisit but it's so much better behaviour-wise let's do that as part of a review before upstreaming.